scraper/github/app/utils/github_requests.py

```diff
@@ -3,132 +3,191 @@
+        @returns : users [json array of users]
+
+        """
+        users = []
+        users_ids = set()
+        per_page = 100
+        last_date = str(date.today())
+        page = 1
+        self.logger.info(f"Starting to fetch users in cameroon/cameroun, max_page={pagination_limit},"
+                         f" per_page={per_page}, last_updated_date={last_date}")
+
+        while pagination_limit:
+            # We set our query
+            # .format() is slow let's use f-string
+            self.logger.info(f"Fetching page={page}, last_date={last_date}")
+            query = f"created:<{last_date}+sort:joined-desc+location:%22cameroon%22+location:%22cameroun%22" \
+                    f"&page={page}&per_page={per_page}"
+
+            # a simple request to the api
+            r = self.session.get(f"{self.GITHUB_API}/search/users?q={query}")
+            # We check the status of the requet and return a predefined error message
+            schk = self.status_check(r)
+            if not schk[0]:
+                # we check if on_pageloaded_error is a function
+                error_code = schk[1]["code"]
+                if error_code == 422:  # we hit 1000 limit, downgrading the date
+                    last_user: dict = users[-1]
+                    last_user = self.get_user(last_user["login"])
+                    updated_at: str = last_user["updated_at"]
+                    if updated_at:
+                        last_date = updated_at.split("T")[0]
+                        page = 1
+                        continue
+
+                if callable(on_pageloaded_error):
+                    on_pageloaded_error(schk[1])
+                else:
+                    self.logger.warning(f"on_pageloaded_error is passed but it is not callable {on_pageloaded_error}")
+                return schk[1]
+
+            items = r.json()["items"]
+            # if we encounter an empty list we break
+            if not len(items):
+                break
+            self.logger.info(f"Found users {[user['login'] for user in items]}")
+
+            unique_items = []
+            for item in items:
+                if item["id"] in users_ids:
+                    continue
+
+                unique_items.append(item)
+                users_ids.add(item["id"])
+
+            if callable(on_pageloaded_success):
+                on_pageloaded_success(unique_items)
+            else:
+                self.logger.warning(f"on_pageloaded_success is passed but it is not callable {on_pageloaded_success}")
+
+            # We append or merge by ensuring uniciting
+            users.extend(unique_items)
+
+            # if we get less than per_page there is no more result
+            if len(items) < per_page:
+                break
+
+            page += 1
+            pagination_limit -= 1  # ensure we respect the pagination
+            # time.sleep(1) # we sleep 1 seconds to no flood the api
```

Comment:
# time.sleep comment is for you, if you decide to limit the speed of the scraper to not flood the api.
the previous comment is to explain the line. someone could not understand the use of the pagination_limit because the scraper can stop itself when they are no more users.